File: 4_multiple_inheritance.cpp
#include<bits/stdc++.h>
using namespace std;
 

class Home {
public:
    string address;

       
     
};

class School {
public:
    int id;
    int level;

    
     
};

class Sports {
public:
    string game;
     
     
};

class Student : public Home, public School, public Sports {  // multiple inheritance 
    public:
    Student(string address, int id, int level,string game){
        this->address = address;
        this->id = id;
        this->level = level;
        this->game = game;
    }
    void display() {
        cout<<address<<" "<<id<<" "<<level<<" "<<game<<endl;
    }
};

int main() {
    Student a1("kishoreganj", 25, 2, "a");
    a1.display();
    return 0;
}

File: 4_multiple_inheritance.py
class Home:
    def __init__(self, address) -> None:
        self.address = address
    

class School:
    def __init__(self, id, level) -> None:
        self.id = id
        self.level = level


class Sports:
    def __init__(self, game) -> None:
        self.game = game


class Student(Home, School,  Sports):  # multiple inheritance 
    def __init__(self, address, id, level, game) -> None:
        School.__init__(self, id, level)
        Sports.__init__(self, game)
        Home.__init__(self, address)

    def display(self):
        print(f"{self.address}' ' {self.id}' '{self.level}' '{self.game} ")





a1 = Student('kishoreganj', 25, 2, 'a')

a1.display()
